samples/src/main/java/com/example/bigquery/AddColumnLoadAppend.java

```
@@ -0,0 +1,102 @@
+    try {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+      TableId tableId = TableId.of(datasetName, tableName);
+      Table table = bigquery.getTable(tableId);
+
+      // Add a new column to a BigQuery table while appending rows via a load job.
+      // 'REQUIRED' fields cannot  be added to an existing schema, so the additional column must be
+      // 'NULLABLE'.
+      Schema newSchema =
+          Schema.of(
+              Field.newBuilder("name", LegacySQLTypeName.STRING)
+                  .setMode(Field.Mode.REQUIRED)
+                  .build(),
+              // Adding below additional column during the load job
+              Field.newBuilder("post_abbr", LegacySQLTypeName.STRING)
+                  .setMode(Field.Mode.NULLABLE)
+                  .build());
+
+      LoadJobConfiguration loadJobConfig =
+          LoadJobConfiguration.builder(tableId, sourceUri)
+              .setFormatOptions(FormatOptions.csv())
+              .setWriteDisposition(WriteDisposition.WRITE_APPEND)
+              .setSchema(newSchema)
+              .setSchemaUpdateOptions(ImmutableList.of(SchemaUpdateOption.ALLOW_FIELD_ADDITION))
+              .build();
+
+      // Create a job ID so that we can safely retry.
+      JobId jobId = JobId.of(UUID.randomUUID().toString());
+      Job loadJob = bigquery.create(JobInfo.newBuilder(loadJobConfig).setJobId(jobId).build());
+      System.out.println(loadJob.getJobId());
+
+      // Load data from a GCS parquet file into the table
+      // Blocks until this load table job completes its execution, either failing or succeeding.
+      Job completedJob = loadJob.waitFor();
+
+      // Check for errors
+      if (completedJob == null) {
+        throw new RuntimeException("Job no longer exists");
```

nit: Probably InvalidStateException would make more sense?

RuntimeException is unchecked, making it a bad choice to suggest to users.

---

InvalidStateException is not available so I will print out the error message instead

---

I think there's a miscommunication on error handling here.

If something goes wrong, we definitely want to throw an exception - but we want a checked exception vs unchecked.

Check out this article for a breakdown of [checked v unchecked](https://howtodoinjava.com/java/exception-handling/checked-vs-unchecked-exceptions-in-java/).

---

Throwing Exception so that it's checked.